host/class/uac/usb_host_uac/uac_host.c

```diff
@@ -1790,6 +1833,70 @@ esp_err_t uac_host_device_open(const uac_host_device_config_t *config, uac_host_
+{
+    UAC_RETURN_ON_INVALID_ARG(uac_dev_handle);
+    UAC_RETURN_ON_INVALID_ARG(config);
+    UAC_RETURN_ON_FALSE(config->iface_num, ESP_ERR_INVALID_ARG, "Invalid interface number");
+    UAC_RETURN_ON_FALSE(s_uac_driver, ESP_ERR_INVALID_STATE, "UAC Driver is not installed");
+    UAC_RETURN_ON_FALSE(vid, ESP_ERR_INVALID_ARG, "Invalid vendor ID");
+    UAC_RETURN_ON_FALSE(pid, ESP_ERR_INVALID_ARG, "Invalid product ID");
+
+    uint32_t timeout_cnt = 0;
+    do {
+        // current time in ticks
+        uint32_t timeout_start = xTaskGetTickCount();
+        // check if the device is already opened
+        *uac_dev_handle = get_interface_by_vid_pid(vid, pid, config->iface_num);
+        if (*uac_dev_handle) {
+            ESP_LOGD(TAG, "Device VID 0x%04X, PID 0x%04X, Interface %d already opened", vid, pid, config->iface_num);
+            return ESP_OK;
+        }
+
+        // check all connected devices
+        int actual_count = 0;
+        uint8_t dev_addr_list[6] = {0};
+        usb_host_device_addr_list_fill(sizeof(dev_addr_list), dev_addr_list, &actual_count);
+        if (actual_count == 0) {
+            if (timeout == 0) {
+                ESP_LOGW(TAG, "No devices connected");
+                return ESP_ERR_NOT_FOUND;
+            }
+        }
+
+        for (int i = 0; i < actual_count; i++) {
+            const usb_device_desc_t *dev_desc = NULL;
+            usb_device_handle_t dev_hdl = NULL;
+            uac_device_t *uac_dev = get_uac_device_by_vid_pid(vid, pid);
+            uac_host_device_config_t config_copy = *config;
+            config_copy.addr = dev_addr_list[i];
+            if (uac_dev) {
+                ESP_LOGD(TAG, "Device VID 0x%04X, PID 0x%04X already opened", vid, pid);
+                return uac_host_device_open(&config_copy, uac_dev_handle);
+            } else {
+                UAC_RETURN_ON_ERROR(usb_host_device_open(s_uac_driver->client_handle, dev_addr_list[i], &dev_hdl), "Unable to open USB device");
+                UAC_RETURN_ON_ERROR(usb_host_get_device_descriptor(dev_hdl, &dev_desc), "Unable to get device descriptor");
+                ESP_LOGD(TAG, "Found Device VID 0x%04X, PID 0x%04X", dev_desc->idVendor, dev_desc->idProduct);
+                if (dev_desc->idVendor == vid && dev_desc->idProduct == pid) {
+                    usb_host_device_close(s_uac_driver->client_handle, dev_hdl);
+                    return uac_host_device_open(&config_copy, uac_dev_handle);
+                }
+                usb_host_device_close(s_uac_driver->client_handle, dev_hdl);
+            }
+        }
+        if (timeout == 0) {
+            ESP_LOGW(TAG, "Device VID 0x%04X, PID 0x%04X not found", vid, pid);
+            return ESP_ERR_NOT_FOUND;
+        }
+        vTaskDelay(pdMS_TO_TICKS(50) > timeout ? timeout : pdMS_TO_TICKS(50));
+        timeout_cnt += (xTaskGetTickCount() - timeout_start);
+    } while (timeout && timeout_cnt < timeout);
```

Comment:
Nitpick, does not have to be changed:
you can use [xTaskCheckForTimeOut](https://www.freertos.org/xTaskCheckForTimeOut.html) for this patterm